helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java

```diff
@@ -176,4 +167,21 @@ protected static RealmAwareZkClient createZkClient(HelixManager manager) {
         new HelixZkClient.ZkConnectionConfig(zkAddress),
         clientConfig.createHelixZkClientConfig().setZkSerializer(new ZNRecordSerializer()));
   }
+
+  private static ScheduledExecutorService createThreadPoolExecutor(HelixManager manager) {
+    RealmAwareZkClient zkClient = createZkClient(manager);
```

Comment:
Could we also set the client config so we don't have infinite operation timeout for this particular zk client?